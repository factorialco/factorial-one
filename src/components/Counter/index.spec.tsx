import { render, screen } from "@testing-library/react-native";
import React from "react";
import { Counter } from ".";

describe("Counter", () => {
  const defaultProps = {
    value: 42,
    types: "default",
  };

  it("Snapshot", () => {
    const { toJSON } = render(<Counter {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot vairan defauly types", () => {
    const { toJSON } = render(<Counter {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot vairan defauly types", () => {
    const { toJSON } = render(<Counter {...defaultProps} type="default" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot vairan bold types", () => {
    const { toJSON } = render(<Counter {...defaultProps} type="bold" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot vairan selected types", () => {
    const { toJSON } = render(<Counter {...defaultProps} type="selected" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot vairan md size", () => {
    const { toJSON } = render(<Counter {...defaultProps} size="md" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot vairan sm size", () => {
    const { toJSON } = render(<Counter {...defaultProps} size="sm" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with required props", () => {
    render(<Counter {...defaultProps} />);

    const number = screen.getByText("42");

    expect(number).toBeDefined();
  });
});
