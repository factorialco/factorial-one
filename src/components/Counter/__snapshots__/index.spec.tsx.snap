// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot 1`] = `
<View
  className="flex items-start"
>
  <View
    className="flex items-center justify-center rounded-xs grow-0 px-0.5 min-w-5 h-5 bg-f1-background-secondary border border-f1-border"
  >
    <Text
      className="text-center text-sm font-medium tabular-nums whitespace-nowrap text-f1-foreground"
    >
      42
    </Text>
  </View>
</View>
`;

exports[`Counter Snapshot vairan bold types 1`] = `
<View
  className="flex items-start"
>
  <View
    className="flex items-center justify-center rounded-xs grow-0 px-0.5 min-w-5 h-5 bg-f1-background-accent-bold"
  >
    <Text
      className="text-center text-sm font-medium tabular-nums whitespace-nowrap text-f1-foreground-inverse"
    >
      42
    </Text>
  </View>
</View>
`;

exports[`Counter Snapshot vairan defauly types 1`] = `
<View
  className="flex items-start"
>
  <View
    className="flex items-center justify-center rounded-xs grow-0 px-0.5 min-w-5 h-5 bg-f1-background-secondary border border-f1-border"
  >
    <Text
      className="text-center text-sm font-medium tabular-nums whitespace-nowrap text-f1-foreground"
    >
      42
    </Text>
  </View>
</View>
`;

exports[`Counter Snapshot vairan defauly types 2`] = `
<View
  className="flex items-start"
>
  <View
    className="flex items-center justify-center rounded-xs grow-0 px-0.5 min-w-5 h-5 bg-f1-background-secondary border border-f1-border"
  >
    <Text
      className="text-center text-sm font-medium tabular-nums whitespace-nowrap text-f1-foreground"
    >
      42
    </Text>
  </View>
</View>
`;

exports[`Counter Snapshot vairan md size 1`] = `
<View
  className="flex items-start"
>
  <View
    className="flex items-center justify-center rounded-xs grow-0 px-0.5 min-w-5 h-5 bg-f1-background-secondary border border-f1-border"
  >
    <Text
      className="text-center text-sm font-medium tabular-nums whitespace-nowrap text-f1-foreground"
    >
      42
    </Text>
  </View>
</View>
`;

exports[`Counter Snapshot vairan selected types 1`] = `
<View
  className="flex items-start"
>
  <View
    className="flex items-center justify-center rounded-xs grow-0 px-0.5 min-w-5 h-5 bg-f1-background-selected-bold outline-f1-border-selected"
  >
    <Text
      className="text-center text-sm font-medium tabular-nums whitespace-nowrap text-f1-foreground-inverse"
    >
      42
    </Text>
  </View>
</View>
`;

exports[`Counter Snapshot vairan sm size 1`] = `
<View
  className="flex items-start"
>
  <View
    className="flex items-center justify-center rounded-xs grow-0 px-0.5 min-w-4 h-4 bg-f1-background-secondary border border-f1-border"
  >
    <Text
      className="text-center text-sm font-medium tabular-nums whitespace-nowrap leading-none py-0.5 text-f1-foreground"
    >
      42
    </Text>
  </View>
</View>
`;
